pkg/network/gateway_lookup_linux.go

```diff
@@ -47,12 +46,13 @@ var gatewayLookupTelemetry = struct {
 	telemetry.NewStatCounterWrapper(gatewayLookupModuleName, "subnet_lookup_errors", []string{"reason"}, "Counter measuring the number of subnet lookup errors"),
 }
 
-type gatewayLookup struct {
-	procRoot            string
+// LinuxGatewayLookup implements a gateway lookup
+// functionality for linux
+type LinuxGatewayLookup struct {
```

Comment:
Can we just call this `gatewayLookup`? There is no non-linux version, and even if there was, this code would be compiled out so that we can also name it the same on windows.